torch/__init__.py

```diff
@@ -185,6 +185,36 @@ class ByteStorage(_C.ByteStorageBase, _StorageBase):
     pass
 
 
+# Numpy array interface, to support `numpy.asarray(tensor) -> ndarray`
+def __array__(self, dtype=None):
+    if dtype is None:
+        return self.cpu().numpy()
+    else:
+        return self.cpu().numpy().astype(dtype, copy=False)
+
+
+# Wrap Numpy array again in a suitable tensor when done, to support e.g.
+# `numpy.sin(tensor) -> tensor` or `numpy.greater(tensor, 0) -> ByteTensor`
+def __array_wrap__(self, array):
+    if array.ndim == 0:
+        if array.dtype.kind == 'b':
+            return bool(array)
+        elif array.dtype.kind in ('i', 'u'):
+            return int(array)
+        elif array.dtype.kind == 'f':
+            return float(array)
+        elif array.dtype.kind == 'c':
+            return complex(array)
+        else:
+            raise RuntimeError('bad scalar {!r}'.format(array))
+    else:
+        if array.dtype == bool:
+            # Workaround, torch has no built-in bool tensor
+            array = array.astype('uint8')
+
+        return from_numpy(array)
```

Comment:
You can add these two to `_TensorBase`

Reply:
Sounds easier, I'll do that.

Reply:
That's not what I meant. Add them as regular methods [here](https://github.com/pytorch/pytorch/blob/master/torch/tensor.py#L8).

Reply:
Ah, alright. Didn't realize that `_TensorBase` was implemented in Python. I'll move the code.